fedimint-server/src/net/p2p_connection.rs

```diff
@@ -15,11 +15,25 @@ use tokio_util::codec::{Framed, LengthDelimitedCodec};
 
 pub type DynP2PConnection<M> = Box<dyn IP2PConnection<M>>;
 
+pub type DynMessage<M> = Box<dyn Message<M>>;
+
+#[async_trait]
+pub trait Message<M>: Send + 'static {
+    async fn read(&mut self) -> anyhow::Result<M>;
+
+    fn into_dyn(self) -> DynMessage<M>
+    where
+        Self: Sized,
+    {
+        Box::new(self)
+    }
+}
+
 #[async_trait]
 pub trait IP2PConnection<M>: Send + 'static {
     async fn send(&mut self, message: M) -> anyhow::Result<()>;
 
-    async fn receive(&mut self) -> anyhow::Result<M>;
+    async fn accept(&mut self) -> anyhow::Result<DynMessage<M>>;
```

Comment:
I don't like the rename, it only makes sense for iroh and made me wonder how this interacts with the old way of making connections for a moment.

We already *accepted* (or opened ourselves) a connection, now we are *receiving* messages. The fact that we are using the stream abstraction of iroh instead of our own framing is an implementation detail that shouldn't leak through this API.